docs/export-map.rst

```diff
@@ -0,0 +1,163 @@
+export-map
+==========
+
+.. dfhack-tool::
+    :summary: Export fortress map tile data to a JSON file
+    :tags: dev map
+
+WARNING - This command will cause the game to freeze for minutes depending on
+map size and options enabled.
+
+Exports the fortress map tile data to a JSON file. (does not include items,
+characters, buildings, etc.) Depending on options enabled, there will be a
+``KEY`` table in the JSON with relevant [number ID] values that match a number
+to their object type.
+
+Usage
+-----
+
+::
+
+    export-map [include|exclude] [<options>]
+
+Examples
+--------
+
+``export-map``
+    Exports the fortress map to JSON with ALL data included
+
+``export-map include -m -s -v``
+    Exports the fortress map to JSON with only materials, shape, and variant
+    data included
+
+``export-map exclude --variant --hidden --light``
+    Exports the fortress map to JSON with variant, hidden, and light data
+    excluded
+
+Required
+--------
+
+When you are using options, you must include one of these settings.
+
+``include``
+    Include only the data listed from options to the JSON (whitelist)
+
+``exclude``
+    Exclude only the data listed from options to the JSON (blacklist)
+
+Options
+-------
+
+``help``, ``--help``
+    Shows the help menu
+
+``-t``, ``--tiletype``
+    The tile material classification [number ID] (AIR/SOIL/STONE/RIVER/etc.)
+
+``-s``, ``--shape``
+    The tile shape classification [number ID] (EMPTY/FLOOR/WALL/STAIR/etc.)
+
+``-p``, ``--special``
+    The tile surface special properties for smoothness [number ID]
+    (NORMAL/SMOOTH/ROUGH/etc.) (used for engraving)
+
+``-v``, ``--variant``
+    The specific variant of a tile that have visual variations [number] (like
+    grass tiles in ASCII mode)
+
+``-h``, ``--hidden``
+    Whether tile is revealed or unrevealed [boolean]
+
+``-l``, ``--light``
+    Whether tile is exposed to light [boolean]
+
+``-b``, ``--subterranean``
+    Whether the tile is considered underground [boolean] (used to determine
+    crops that can be planted underground)
+
+``-o``, ``--outside``
+    Whether the tile is considered “outside”. [boolean] (used by weather effects
+    to trigger on outside tiles)
+
+``-a``, ``--aquifer``
+    Whether the tile is considered an aquifer [number ID] (NONE/LIGHT/HEAVY)
+
+``-m``, ``--material``
+    The material inside the tile [number ID] (IRON/GRANITE/CLAY/
+    TOPAZOLITE/BLACK_OPAL/etc.) (will return nil if the tile is empty)
+
+``-u``, ``--underworld``
+    Whether the underworld z-levels will be included
+
+``-e``, ``--evilness``
+    Whether the evilness value will be included in MAP_SIZE table. This only
+    checks the value of the center map tile at ground level and will ignore
+    biomes at the edges of the map.
+
+JSON DATA
```

Comment:
What are the client requirements that led to this output format? The choices made here seem strange to me, and I could use some context.